R/codemeta_readme.R

```diff
@@ -1,153 +1,164 @@
+# guess_ci ---------------------------------------------------------------------
+
 ## Based on CI badges, not config files
 # only Travis, Appveyor and Circle CI
 # also uses code coverage
 guess_ci <- function(readme) {
 
-  badges <- extract_badges(readme)
-  ci_badge <- badges[grepl("travis", badges$link)|
-                       grepl("appveyor", badges$link)|
-                       grepl("circleci", badges$link)|
-                       grepl("codecov", badges$link)|
-                       grepl("coveralls", badges$link),]
-  if (!is.null(ci_badge)) {
-    ci_badge$link
-  } else {
-    NULL
-  }
-
+  get_badge_links_matching(readme, "travis|appveyor|circleci|codecov|coveralls")
 }
 
+# guess_devStatus --------------------------------------------------------------
+
 ## Either repostatus.org or lifecycle badge
 guess_devStatus <- function(readme) {
+
+  get_badge_links_matching(readme, "repostatus\\.org|lifecycle")
+}
+
+# get_badge_links_matching -----------------------------------------------------
+get_badge_links_matching <- function(readme, pattern) {
+
   badges <- extract_badges(readme)
-  status_badge <- badges[grepl("repostatus.org", badges$link)|
-                           grepl("lifecycle", badges$link),]
-  if (!is.null(status_badge)) {
-    if(nrow(status_badge) >0){
-      status_badge$link
-    }
-  } else {
-    NULL
+
+  if (length(links <- grep(pattern, badges$link, value = TRUE))) {
+
+    links
   }
+  # else NULL implicitly
+}
 
+# get_pkg_name -----------------------------------------------------------------
+get_pkg_name <- function(entry) {
 
+  if (is.null(result <- entry$pkgname)) "" else result
 }
 
+# .ropensci_reviews ------------------------------------------------------------
 # looks for a rOpenSci peer review badge
+.ropensci_reviews <- function() {
 
-get_pkg_name <- function(entry){
-  if(is.null(entry$pkgname)){
-    ""
-  }else{
-    entry$pkgname
-  }
-}
+  url_onboarded_json <- "https://badges.ropensci.org/json/onboarded.json"
+
+  reviews <- jsonlite::read_json(url_onboarded_json)
 
-.ropensci_reviews <- function(){
-  reviews <-  jsonlite::read_json("https://badges.ropensci.org/json/onboarded.json")
-  reviews <- tibble::tibble(review = purrr::map_dbl(reviews, "iss_no"),
-                            package = purrr::map_chr(reviews, get_pkg_name))
+  reviews <- tibble::tibble(
+    review = purrr::map_dbl(reviews, "iss_no"),
+    package = purrr::map_chr(reviews, get_pkg_name)
+  )
```

Comment:
is something returned?

Reply:
I assume that reviews is to be returned. I will remove the assignment.

Reply:
👌